rllib/agents/dqn/r2d2.py

```diff
@@ -131,14 +127,14 @@ def validate_config(self, config: TrainerConfigDict) -> None:
         # Call super's validation method.
         super().validate_config(config)
 
-        if config["replay_sequence_length"] != -1:
+        if config["replay_buffer_config"]["replay_sequence_length"] != -1:
```

Comment:
With your improved ReplayBuffer API I wonder if it makes sense to keep around `R2D2` as a separate agent from DQN. AFAIK the only difference is the use of the TD error weighting function `h` and LSTM burn-in. If you plug these options into the `DQN` config, would you get distributed R2D2 for free via `APEX`?

Not worth doing in this PR, but might be worth doing after this is merged.

Reply:
I have a very similar opinion.
I believe we should keep our algorithms section slimmer and provide R2D2 as an example script.
Same goes for RNNSAC.